### DISCUSSION TITLE
Discard log macros and provide my own MacroMetadata?

### DISCUSSION THREAD

**Discussion by kristenmwegner** [odygrd/quill#396] [Q&A]
I like this library, but I have a situation in which I cannot use the predefined logging macros as they are. I need to construct my own quill::MacroMetadata elsewhere, on my own terms, and provide it at the time of the log() call. Unfortunately, in the current implementation the metadata are obtained at compile-time through the log method template argument, so there is no way to do this without writing a completely new log() method. Any suggestions about how to approach this?

**Comment by odygrd** [comment]
Creating a `static constexpr Metadata` object yourself and passing it to the `log()` method is a viable alternative to using macros. Metadata is always compile-time in this library for performance reasons. 
If you could provide a code sample illustrating your requirements, it would be easier for me to further assist
